src/utils/dom-tranform-stream.js

```diff
@@ -147,9 +153,19 @@ export class DomTransformStream {
         this.currentChunkTransferPromise_ = null;
         const targetBody = resolvedElements[0];
         removeNoScriptElements(dev().assertElement(this.detachedBody_));
-        while (this.detachedBody_.firstChild) {
+
+        // Always leave one node that may get updated on next chunk.
+        while (this.detachedBody_.children.length > 1) {
           targetBody.appendChild(this.detachedBody_.firstChild);
         }
+
+        // Transfer the rest of doc, including any non-element nodes that may
+        // be trailing.
+        if (this.onEndCalled_) {
```

Comment:
I think this logic is ok. But I'm still convinced it's needed. Is there a test that demonstrates a failure here? From what I remember, a parser should still keep updating an element, even if it has been transferred into a different document.

Reply:
Tried to write a unit test and you are right, the `document.write` call is smart enough to find the transferred element. 

I see many JSON parse errors in our logs for this experiment. I am also able to repro with a test page that returns a very slow response in small chunks. After your feedback I now think the error is not that the transferred element does not get updated, but that the update happens after build/layout get called when it is too late.

Reply:
Hmm. The text nodes are a very good point. I do think this PR helps, though the cost is pretty high. In a special case where a body contains only a single element, e.g. `<body><main>...</main></body>`, this will be equivalent to turning off streaming. 

The reason I'm bringing this up is because the main AMP case has the same problem and almost the same logic. What we do there is to look for the next sibling before considering an element complete. That work is assisted by the ampdoc's `whenReady` signal. However, I'm noticing that this is not necessarily true for FIE embeds in the current resources-impl. The good news, however, I believe this should be addressed by the https://github.com/ampproject/amphtml/issues/31246 launch. So let's see if want to "cancel" streaming for now or wait for the FIE inabox-resources to fully launch.